### PR TITLE
override default CORS handler for httprouter

### PR DESCRIPTION
**Pull request by Alma-media**
Related to https://github.com/jexia/semaphore/issues/74

Includes support for:
- `Access-Control-Allow-Origin` 
- `Access-Control-Request-Method`
- `Access-Control-Request-Headers`

Other changes:
- introduce `ListenerOption`

### CODE REVIEW

File: cmd/semaphore/config/arguments.go [hunk 1]
```diff
@@ -60,8 +62,26 @@ func ConstructArguments(params *Semaphore) ([]config.Option, error) {
 		arguments = append(arguments, semaphore.WithServices(protobuffers.ServiceResolver(params.Protobuffers, path)))
 	}
 
+	log.Println()
+	log.Println()
+	log.Println(params.HTTP)
+	log.Println()
+	log.Println()
+
```
Comment (jeroenrinzema) [inline cmd/semaphore/config/arguments.go:70]: please remove the debug logs.

File: pkg/transport/http/listener_options_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
+package http
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func Test_NewListenerOptions(t *testing.T) {
```
Comment (jeroenrinzema) [inline pkg/transport/http/listener_options_test.go:10]: Let's keep the test names consistent? I have not yet used underscores in test names inside this project.

File: cmd/semaphore/config/arguments.go [hunk 3]
```diff
@@ -60,8 +62,26 @@ func ConstructArguments(params *Semaphore) ([]config.Option, error) {
...
+		arguments = append(
+			arguments,
+			semaphore.WithListener(
+				http.NewListener(
+					params.HTTP.Address,
+					http.WithOrigins(params.HTTP.Origin),
+					http.WithReadTimeout(params.HTTP.ReadTimeout),
+					http.WithWriteTimeout(params.HTTP.WriteTimeout),
+					http.WithKeyFile(params.HTTP.KeyFile),
+					http.WithCertFile(params.HTTP.CertFile),
```
Comment (jeroenrinzema) [inline cmd/semaphore/config/arguments.go:81]: The options could be passed but are not yet configurable. Please include these options inside the HCL provider and parse the options.

File: cmd/semaphore/config/arguments.go (hunk 1 above)
Comment (Alma-media) [reply cmd/semaphore/config/arguments.go:70]: oops

File: cmd/semaphore/config/arguments.go (hunk 3 above)
Comment (Alma-media) [reply cmd/semaphore/config/arguments.go:81]: it was already done :thinking:

Comment (jeroenrinzema) [reply cmd/semaphore/config/arguments.go:81]: HCL syntax is defined over here:
https://github.com/jexia/semaphore/blob/master/pkg/providers/hcl/intermediate.go#L31

Options fetched over here:
https://github.com/jexia/semaphore/blob/master/pkg/providers/hcl/hcl.go#L106

Comment (jeroenrinzema) [reply cmd/semaphore/config/arguments.go:73]: My bad! I thought I did not see the change.

**Pull request merged**